src/settings/settings.rs

```diff
@@ -0,0 +1,55 @@
+use config::{Config, Environment, File};
+use serde_derive::{Deserialize};
+use std::{env, path::PathBuf, collections::HashMap};
+use lazy_static::lazy_static;
+
+#[derive(Deserialize, Clone, Hash)]
+#[allow(unused)]
+pub struct Database {
+    pub folder: String,
+}
+
+#[derive(Deserialize, Clone, Hash)]
+#[allow(unused)]
+pub struct Suma {
+    pub url: String,
+    pub user: String,
+    pub password: String,
+}
+
+#[derive(Deserialize, Clone, Hash)]
+#[allow(unused)]
+pub struct Server {
+    pub listen: String,
+}
+
+#[derive(Deserialize, Clone, Hash)]
+#[allow(unused)]
+pub struct Settings {
+    pub database: Database,
+    pub suma: Suma,
+    pub server: Server,
+}
+
+
+impl Settings {
+    pub fn new() -> Self {
```

Comment:
I would use `load` as the name for reading a setting from the environment (or maybe `read`). `new` is correct in the sense, that Rust uses this convenience to initialise new structs with `new`, but from the operation it is actually a reading or loading from the environment. I would prefer the later, so it is clear what it does.